### PR TITLE
CIS-1166 Improve logging

### PR DESCRIPTION
**Pull request by tbarbugli**
- Set sample app log level to .warning
- Log error responses from API with .warning level]
- Skip useless warning message when connecting

### CODE REVIEW

File: Sources/StreamChat/Workers/ChatClientUpdater.swift [hunk 1]
```diff
@@ -159,7 +159,7 @@ class ChatClientUpdater {
             return
         }
 
-        guard client.connectionId != nil else {
+        if client.connectionId == nil && source == .userInitiated {
```
Comment (dmigach) [inline Sources/StreamChat/Workers/ChatClientUpdater.swift:162]: I'm wondering if going past this expression for system initiated calls and calling `client.completeConnectionIdWaiters(connectionId: nil)` might cause any unpleasant side-effects, but probably it's all right

**Review by dmigach — APPROVED** [review]

File: Sources/StreamChat/Workers/ChatClientUpdater.swift (hunk 1 above)
Comment (tbarbugli) [reply Sources/StreamChat/Workers/ChatClientUpdater.swift:162]: you are 100% right, fixing this

File: DemoApp/DemoAppCoordinator.swift [hunk 2]
```diff
@@ -72,6 +72,7 @@ final class DemoAppCoordinator: NSObject, UNUserNotificationCenterDelegate {
         // Create a token
         let token = try! Token(rawValue: userCredentials.token)
         
+        LogConfig.level = .warning
```
Comment (b-onc) [inline DemoApp/DemoAppCoordinator.swift:75]: This is set in line 21

**Review by b-onc — COMMENTED** [review]
Redundant LogConfig.level setting in DemoApp
OOT, LGTM

**Pull request merged**